### ISSUE TITLE
Double-tap actions for messages

### ISSUE THREAD

**Issue by starspritechippy** [Aliucord/plugin-requests#2]
### Description
A plugin that allows message interactions when double tapping them.

### More info
Message interactions include editing them, replying to them, deleting them. Which action is used in which context (whether it's your own message or someone else's) *could* be configurable.

---

- [x] I checked if this plugin wasn't already requested.

This was already requested in the channel, I'm more or less just adding this so that it's more organized here.

**Comment by Swishilicous** [comment]
i think triple tap would be more mobile friendly but ig that could just be a seperate option 🤷‍♀️

**Comment by starspritechippy** [comment]
I believe @Litleck mentioned, that they were interested in giving this a shot.

**Comment by D3C0D3R5** [comment]
Whoever makes this. Please be sure to add a "Double Tap To Auto-React With Emotes" and it would be custom chosen emotes that are customizable from within the "Plugin Settings".

**Comment by vy-storm** [comment]
I'd like to suggest the following interaction types:
- Hold (override default)
- Double tap
- Triple tap
- Tap and hold
- Double tap and hold

Each of these twice, one for your own messages and one for others.  These interactions could then be set to any function accessable from the pop-up menu or simply disabled altogether.

This level of depth would be a dream come true for power users, potentially even eliminating the need to use the pop-up menu whatsoever for some people.

**Comment by Vendicated** [comment]
This is somewhat implemented in [TapTap](https://github.com/Vendicated/AliucordPlugins/blob/main/TapTap). It makes it so that double tapping your own message starts editing it and double tapping someone else's message replies to it

Changing actions is not possible as of now so you're stuck with those actions. Perhaps in the future i could add some sort of toggle (no idea where yet, it should be accessible so somewhere in the chat area might be good?) to change the action that is done on the fly (reply, delete, pin, etc)

I experimented with triple tap but it does not seem very viable to me. The triple tap window needs to be big enough (~400ms) or you cannot tap quickly enough and this means that double taps have to be delayed for 400ms too to make sure there's not a third tap, so everything becomes sloooow

**Comment by vy-storm** [comment]
> This is partially implemented in [TapTap](https://github.com/Vendicated/AliucordPlugins/blob/main/TapTap). It makes it so that double tapping your own message starts editing it and double tapping someone else's message replies to it
> 
> Changing actions is not possible as of now so you're stuck with those actions. Perhaps in the future i could add some sort of toggle (no idea where yet, it should be accessible so somewhere in the chat area might be good?) to change the action that is done on the fly (reply, delete, pin, etc)
> 
> I experimented with triple tap but it does not seem very viable to me. The triple tap window needs to be big enough (~400ms) or you cannot tap quickly enough and this means that double taps have to be delayed for 400ms too to make sure there's not a third tap, so everything becomes sloooow
> 
> The combinations described in the reply above would be pretty complex to implement but definitely possible

Yeah that's fair about the triple tap, I experienced a similar thing in an app that lets you trigger actions on your phone by tapping the back multiple times.  The latency does make triple tap rather unideal.

What of just tap and hold instead of triple tap then mayhaps along with and the regular hold override?

Also interesting idea of changing these on the fly, I was just thinking of having all the switches in the plugin settings, but a way to do that on the go method might be to hold down on options in the menu?  That could open a window where you could bind that action to a tap trigger?  Wouldn't work very well with rebinding the original hold action tho

**Comment by lightsideluc** [comment]
Given the difficulties with triple tap, would it instead be viable to have a Hold+Tap (two fingers, one to hold the message and a second to tap/double tap it) function?
